src/components/TodosContext/TodosContext.tsx

```diff
@@ -0,0 +1,243 @@
+  useState,
+} from 'react';
+import { Todo } from '../../types/Todo';
+import {
+  createTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from '../../api/todos';
+import { FilterType } from '../../types/EnumFilterType';
+
+export const USER_ID = 11537;
+
+interface Context {
+  todos: Todo[],
+  visibleTodos: Todo[],
+  isLoading: { [key: number]: boolean },
+  setLoading: (key: number, value: boolean) => void,
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>
+  setFilterType: (value: FilterType) => void,
+  tempTodo: Todo | null,
+  setTempTodo: (todo: Todo | null) => void,
+  errorMessage: string,
+  setErrorMessage: (value: string) => void,
+  addTodo: (value: string) => Promise<Todo>
+  updateAllCheckbox: () => void,
+  hangleDeleteTodo: (todoId: number) => void,
+  updateInputCheckbox: (todo: Todo) => void,
+  deleteComplitedTodos: () => void,
+}
+
+export const TodosContext = React.createContext<Context>({
+  todos: [],
+  setTodos: () => { },
+  visibleTodos: [],
+  isLoading: {},
+  setLoading: () => { },
+  setFilterType: () => { },
+  tempTodo: null,
+  setTempTodo: () => { },
+  errorMessage: '',
+  setErrorMessage: () => { },
+  addTodo: () => Promise.resolve({} as Todo),
+  updateAllCheckbox: () => { },
+  hangleDeleteTodo: () => { },
+  updateInputCheckbox: () => { },
+  deleteComplitedTodos: () => { },
+});
+
+interface Props {
+  children: ReactNode;
+}
+
+export const TodosProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const [filterType, setFilterType] = useState(FilterType.All);
+  const [isLoading, setIsLoading] = useState<{ [key: string]: boolean }>({});
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const timerErrorMessage = useRef<number>(0);
+  const incompleteTodos = todos.some(({ completed }) => completed)
+    ? todos.filter(({ completed }) => !completed)
+    : todos;
+
+  useEffect(() => {
+    if (timerErrorMessage.current) {
+      window.clearTimeout(timerErrorMessage.current);
+    }
+
+    timerErrorMessage.current = window.setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  }, [errorMessage]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((response) => {
+        setTodos(response);
+      })
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+      });
+  }, []);
+
+  const setLoading = (key: number, value: boolean) => {
+    setIsLoading((prevLoadingKeys) => ({
+      ...prevLoadingKeys,
+      [key]: value,
+    }));
+  };
+
+  const addTodo = (prerapedTitle: string): Promise<Todo> => {
+    return createTodo({
+      title: prerapedTitle,
+      completed: false,
+      userId: USER_ID,
+    })
+      .then(newPost => {
+        setTodos((prevTodos) => [...prevTodos, newPost]);
+
+        return newPost;
+      })
+      .catch(() => {
+        setErrorMessage('Unable to add a todo');
+        throw new Error();
```

Comment:
The error text must be passed to the Error class construct, otherwise it makes little sense.